android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt

```diff
@@ -210,4 +222,77 @@ class ProfileViewModelTest : RobolectricTest() {
       profileViewModel.onEvent(event)
     }
   }
+
+  @Test
+  fun testManagingEntityShowsProfileBottomSheetFragment() = runTest {
+    val navController = mockk<NavController>()
+    val event = mockk<ProfileEvent.OverflowMenuClick>()
+    val fragmentManager = mockk<FragmentManager>()
+    val fragmentManagerTransaction = mockk<FragmentTransaction>()
+    val overflowMenuItemConfig =
+      OverflowMenuItemConfig(
+        id = 1,
+        title = "open profile bottom sheet",
+        confirmAction = false,
+        icon = null,
+        titleColor = BLACK_COLOR_HEX_CODE,
+        backgroundColor = null,
+        visible = "true",
+        showSeparator = false,
+        enabled = "true",
+        actions = emptyList(),
+      )
+    val group =
+      Group().apply { managingEntity = managingEntity.apply { reference = "patient/1424251" } }
+    val managingEntityResource = mockk<Group.GroupMemberComponent>()
+    val profileBottomSheetFragment = mockk<ProfileBottomSheetFragment>()
+    val activity = mockk<AppCompatActivity>()
+
+    val viewModel =
+      ProfileViewModel(
+        registerRepository,
+        configurationRegistry,
+        dispatcherProvider,
+        fhirPathDataExtractor,
+        resourceDataRulesExecutor,
+      )
+
+    val managingEntityConfig =
+      ManagingEntityConfig(
+        nameFhirPathExpression = "name",
+        eligibilityCriteriaFhirPathExpression = "criteria",
+        resourceType = ResourceType.Patient,
+        dialogTitle = "Change Managing Entity",
+        dialogWarningMessage = "Warning",
+        dialogContentMessage = "Select a new managing entity",
+        noMembersErrorMessage = "No members found",
+        managingEntityReassignedMessage = "Managing entity reassigned",
+      )
+
+    coEvery { registerRepository.loadResource<Group>("group1") } returns group
+    coEvery { group.member } returns listOf(managingEntityResource)
+    every { managingEntityResource.id } returns "entity1"
+    every {
+      fhirPathDataExtractor.extractValue(
+        managingEntityResource,
+        "name",
+      )
+    } returns "memebr 1"
+    every { fhirPathDataExtractor.extractValue(managingEntityResource, any()) } returns "true"
```

Comment:
Is it possible to avoid using mocks here? `fhirPathDataExtractor` is simply a utility class.